packages/platform/atoms/connect/apple/AppleConnect.tsx

```diff
@@ -0,0 +1,100 @@
+import {
+  Dialog,
+  DialogTrigger,
+  DialogContent,
+  DialogTitle,
+  DialogHeader,
+  DialogDescription,
+} from "@/components/ui/dialog";
+import type { FC } from "react";
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+
+import { Button, Form, PasswordField, TextField } from "@calcom/ui";
+
+import { AtomsWrapper } from "../../src/components/atoms-wrapper";
+import { cn } from "../../src/lib/utils";
+import type { OAuthConnectProps } from "../OAuthConnect";
+
+export const AppleConnect: FC<Partial<Omit<OAuthConnectProps, "redir">>> = ({
+  label = "Connect Apple Calendar",
+  className,
+}) => {
+  const [isDialogOpen, setIsDialogOpen] = useState(false);
+  const displayedLabel = label;
+  const isDisabled = false;
+
+  const form = useForm({
+    defaultValues: {
+      username: "",
+      password: "",
+    },
+  });
+
+  return (
+    <AtomsWrapper>
+      <Dialog open={isDialogOpen}>
+        <DialogTrigger>
+          <Button
+            StartIcon="calendar-days"
+            color="primary"
+            disabled={isDisabled}
+            className={cn("", className, isDisabled && "cursor-not-allowed", !isDisabled && "cursor-pointer")}
+            onClick={() => setIsDialogOpen(true)}>
+            {displayedLabel}
+          </Button>
+        </DialogTrigger>
+        <DialogContent>
+          <DialogHeader>
+            <DialogTitle>Connect to Apple Server</DialogTitle>
+            <DialogDescription>
+              Generate an app specific password to use with Cal.com at{" "}
+              <span className="font-bold">https://appleid.apple.com/account/manage</span>. Your credentials
+              will be stored and encrypted.
+            </DialogDescription>
+          </DialogHeader>
+          <Form
+            form={form}
+            handleSubmit={(values) => {
+              console.log(values);
+              setIsDialogOpen(false);
+              form.reset();
```

Comment:
call the mutation from useSaveCalendarCredentials here, reset the form and close the dialog on success, leave dialog open and display error onErrror